src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -262,7 +272,7 @@ public static bool IsWindowsAzureManagedHosting(this IEnvironment environment)
         /// <returns><see cref="true"/> if running in a Linux Consumption App Service app; otherwise, false.</returns>
         public static bool IsAnyLinuxConsumption(this IEnvironment environment)
         {
-            return environment.IsLinuxConsumptionOnAtlas() || environment.IsLinuxConsumptionOnLegion();
+            return (environment.IsLinuxConsumptionOnAtlas() || environment.IsLinuxConsumptionOnLegion()) && !environment.IsManagedAppEnvironment();
```

Comment:
There isn't an existing test for this method, but please add one that tests the existing logic as well as your new logic